hilti/include/ast/scope-lookup.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2020 by the Zeek Project. See LICENSE for details.
+//
+// Functionality factored out from scope.h to avoid header loops.
+
+#pragma once
+
+#include <optional>
+#include <string>
+
+#include <hilti/ast/declarations/type.h>
+#include <hilti/ast/id.h>
+#include <hilti/ast/module.h>
+#include <hilti/ast/node_ref.h>
+#include <hilti/ast/type.h>
+#include <hilti/base/visitor-types.h>
+
+namespace hilti::scope {
+
+namespace detail {
+/** Internal backend to `hilti::lookupID()`. */
+extern std::pair<bool, Result<std::pair<NodeRef, ID>>> lookupID(const ID& id, const Node& n);
```

Comment:
No need to declare this `extern`.
```suggestion
std::pair<bool, Result<std::pair<NodeRef, ID>>> lookupID(const ID& id, const Node& n);
```